ITU-BDSA2024-GROUP6/Chirp
# End-To-End Test Implementation

This commit introduces end-to-end testing capabilities for the Chirp project. It sets up a new xUnit test class specifically for testing the public and user timelines, allowing us to verify the rendering of chirps on the web interface.

Changes include:

- Implement E2ETest class with two test cases:
   - Verify the public timeline page contains multiple chirps and is accessible.
   - Verify a user's timeline displays the correct author and associated chirps.
- Utilize HttpClient to send GET requests and validate the HTML response content.
- Ensure proper assertions to check for the presence of chirps and authors.

To run the tests:
- Start the Chirp server: dotnet run in the Chirp.Razor directory
- Run tests: dotnet test in the test/Chirp.E2ETest directory

Note: Ensure the BaseUrl in E2ETest.cs matches your local server address.

These tests provide a solid foundation for validating the read functionality of the Chirp application as we continue to develop and enhance its features.
